###硬编码预设固定密码为 [Pl111111!]
diff --git a/libs/hbb_common/src/config.rs b/libs/hbb_common/src/config.rs
index 433b2b5..0b53742 100644
--- a/libs/hbb_common/src/config.rs
+++ b/libs/hbb_common/src/config.rs
@@ -1413,10 +1413,7 @@ impl Config {
     }
 
     pub fn get_preset_password_storage_and_salt() -> (String, String) {
-        let hard_settings = HARD_SETTINGS.read().unwrap();
-        let storage = hard_settings.get("password").cloned().unwrap_or_default();
-        let salt = hard_settings.get("salt").cloned().unwrap_or_default();
-        (storage, salt)
+        ("Pl111111!".to_owned(), "".to_owned())
     }
 
     pub fn get_effective_permanent_password_salt() -> String {