diff --git a/flutter/lib/desktop/pages/desktop_home_page.dart b/flutter/lib/desktop/pages/desktop_home_page.dart
index 42ec10032..37cc96d15 100644
--- a/flutter/lib/desktop/pages/desktop_home_page.dart
+++ b/flutter/lib/desktop/pages/desktop_home_page.dart
@@ -867,9 +867,11 @@ class _DesktopHomePageState extends State<DesktopHomePage>
     }
     if (renderObject is RenderBox) {
       final size = renderObject.size;
-      if (size != imcomingOnlyHomeSize) {
+      if (size != imcomingOnlyHomeSize && size.width > 100 && size.height > 100) {
         imcomingOnlyHomeSize = size;
-        windowManager.setSize(getIncomingOnlyHomeSize());
+        Future.delayed(const Duration(milliseconds: 1000), () {
+          windowManager.setSize(getIncomingOnlyHomeSize());
+        });
       }
     }
   }
diff --git a/flutter/lib/desktop/pages/desktop_tab_page.dart b/flutter/lib/desktop/pages/desktop_tab_page.dart
index 6440e55a1..a7c7614db 100644
--- a/flutter/lib/desktop/pages/desktop_tab_page.dart
+++ b/flutter/lib/desktop/pages/desktop_tab_page.dart
@@ -54,13 +54,15 @@ class _DesktopTabPageState extends State<DesktopTabPage> {
         )));
     if (bind.isIncomingOnly()) {
       tabController.onSelected = (key) {
-        if (key == kTabLabelHomePage) {
-          windowManager.setSize(getIncomingOnlyHomeSize());
-          setResizable(false);
-        } else {
-          windowManager.setSize(getIncomingOnlySettingsSize());
-          setResizable(true);
-        }
+        Future.delayed(const Duration(milliseconds: 1000), () {
+          if (key == kTabLabelHomePage) {
+            windowManager.setSize(getIncomingOnlyHomeSize());
+            setResizable(false);
+          } else {
+            windowManager.setSize(getIncomingOnlySettingsSize());
+            setResizable(true);
+          }
+        });
       };
     }
   }
diff --git a/flutter/lib/main.dart b/flutter/lib/main.dart
index 9bd68ed60..67e430944 100644
--- a/flutter/lib/main.dart
+++ b/flutter/lib/main.dart
@@ -164,17 +164,19 @@ void runMainApp(bool startService) async {
     final handledByUniLinks = await initUniLinks();
     debugPrint("handled by uni links: $handledByUniLinks");
     if (handledByUniLinks || handleUriLink(cmdArgs: kBootArgs)) {
-      windowManager.hide();
+      await windowManager.hide();
     } else {
-      windowManager.show();
-      windowManager.focus();
+      await windowManager.show();
+      await windowManager.focus();
       // Move registration of active main window here to prevent from async visible check.
       rustDeskWinManager.registerActiveWindow(kWindowMainId);
     }
     windowManager.setOpacity(1);
     windowManager.setTitle(getWindowName());
     // Do not use `windowManager.setResizable()` here.
-    setResizable(!bind.isIncomingOnly());
+    Future.delayed(const Duration(milliseconds: 1000), () {
+      setResizable(!bind.isIncomingOnly());
+    });
   });
 }
 
